Revert #77 (#13176)
NixOS/nix: doc/manual/rl-next/revert-77.md

---
synopsis: Revert incomplete closure mixed download and build feature
issues: [77, 12628]
prs: [13176]
---

Since Nix 1.3 (299141ecbd08bae17013226dbeae71e842b4fdd7 in 2013) Nix has attempted to mix together upstream fresh builds and downstream substitutions when remote substuters contain an "incomplete closure" (have some store objects, but not the store objects they reference).
This feature is now removed.

Worst case, removing this feature could cause more building downstream, but it should not cause ought failures, since this is not happening for opaque store objects that we don't know how to build if we decide not to substitute.
In practice, however, we doubt even the more building is very likely to happen.
Remote stores that are missing dependencies in arbitrary ways (e.g. corruption) doesn't seem to be very common.

> On the contrary, when remote stores fail to implement the closure property (have object -> have all its dependencies), it is usually an *intentional* choice on the remote stores part, because it wishes to serve as an "overlay" store over another store, such as `https://cache.nixos.org`.

Author: ```suggestion
On the contrary, when remote stores fail to implement the closure property, it is usually an *intentional* choice on the part of the remote store, because it wishes to serve as an "overlay" store over another store, such as `https://cache.nixos.org`.
```

> On the contrary, when remote stores fail to implement the [closure property](@docroot@/store/store-object.md#closure-property), it is usually an *intentional* choice on the part of the remote store, because it wishes to serve as an "overlay" store over another store, such as `https://cache.nixos.org`.

Author: OK it's a link now. Thank you @Mic92 for merging the other PR so quickly!
